src/helpers/getFactionWeights.js

```diff
@@ -0,0 +1,18 @@
+import { BRAWLS } from '../constants/brawl'
+import { getLongFaction } from '../helpers/encoding'
+
+export default modifier => {
+  const factions = ['winter', 'ironclad', 'shadowfen', 'swarm']
+
+  if (modifier === 'NONE')
+    return factions.map(faction => ({ id: faction, weight: 1 }))
+
+  const brawlFaction = getLongFaction(
+    BRAWLS.find(brawl => brawl.id === modifier).cardId.slice(0, 1)
+  )
```

Comment:
Interesting! I like the way you did it; you’re so resourceful. I wonder whether it is semantically correct though, since the `cardId` in the `BRAWLS` constant is mostly used for illustration purpose, but I guess it’s good for now.

Reply:
I wanted to use the recommended faction but I guess that attribute is gone 😢

Reply:
It is. 😅